File: server/src/main/java/edu/cnm/deepdive/coffeeshop/repository/InterestRepository.java
package edu.cnm.deepdive.coffeeshop.repository;

import edu.cnm.deepdive.coffeeshop.model.entity.Interest;
import edu.cnm.deepdive.coffeeshop.model.entity.Profile;
import java.util.List;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;

public interface InterestRepository extends JpaRepository<Interest, UUID> {

  // TODO: 7/17/26 Add additional crud methods as needed.

  List<Interest> getAllByOrderByCategoryAsc();
}

File: app/src/main/java/edu/cnm/deepdive/coffeeshop/repository/InterestRepository.kt
package edu.cnm.deepdive.coffeeshop.repository

import edu.cnm.deepdive.coffeeshop.model.domain.Interest
import edu.cnm.deepdive.coffeeshop.model.dto.openapi.InterestDto
import edu.cnm.deepdive.coffeeshop.service.openapi.InterestApi
import jakarta.inject.Inject
import java.util.UUID
import jakarta.inject.Singleton

@Singleton
class InterestRepository @Inject constructor(private val interestApi: InterestApi) {

    suspend fun getInterests(): List<Interest> {
        val response = interestApi.listInterests()
        return if (response.isSuccessful) {
            response.body()?.map { it.toInterest() } ?: emptyList()
        } else {
            throw ServiceException(response.errorBody()?.string() ?: "Unknown error")
        }
    }

    suspend fun getInterest(id: UUID): Interest{
       return interestApi.getInterestById(id).let {
           if (it.isSuccessful) {
               it.body()?.toInterest() ?: throw ServiceException("Unknown error")
           } else {
               throw ServiceException(it.errorBody()?.string() ?: "Unknown error")
           }
       }
    }

    private fun InterestDto.toInterest(): Interest =
        Interest(
            id = this.id,
            category = this.category,
        )
}